internal/git/git.go

```diff
@@ -91,82 +91,112 @@ func (c *Client) getRef(branch string, allowTags bool) (name string, refType str
 	return "", "", fmt.Errorf("error parsing git tag name: git repo at %s must be checked out to a valid branch or tag, or --branch option must be set", c.workspace)
 }
 
-func (c *Client) branchName() (string, error) {
-	/* #nosec */
-	cmd := exec.Command("git", "-C", c.workspace, "rev-parse", "--abbrev-ref", "HEAD")
-	out, err := cmd.CombinedOutput()
+func (c *Client) branchName() (name string, err error) {
+	repo, err := git.PlainOpen(c.workspace)
+	if err != nil {
+		return name, err
```

Comment:
Isn't name undefined at this point? Or does go somehow know to magically make this the empty string. Same question for all the updated methods here that return tuples

Reply:
yep, it will just be an empty string

TIL, you don't even have to list the params in return statement :heart: https://go.dev/tour/basics/7

Reply:
(not going to update these function because they are a but longer/more complex than the example given)